docs/features/nullable-metadata.md

```diff
@@ -0,0 +1,189 @@
+Nullable Metadata
+=========
+The following describes the representation of nullable annotations in metadata.
+
+## NullableAttribute
+Type references are annotated in metadata with a `NullableAttribute`.
+
+```C#
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(
+        AttributeTargets.Class |
+        AttributeTargets.Event |
+        AttributeTargets.Field |
+        AttributeTargets.GenericParameter |
+        AttributeTargets.Parameter |
+        AttributeTargets.Property |
+        AttributeTargets.ReturnValue,
+        AllowMultiple = false)]
+    public sealed class NullableAttribute : Attribute
+    {
+        public readonly byte[] Flags;
+        public NullableAttribute(byte flag)
+        {
+            Flags = new byte[] { flag };
+        }
+        public NullableAttribute(byte[] flags)
+        {
+            Flags = flags;
+        }
+    }
+}
+```
+
+The `NullableAttribute` type is for compiler use only - it is not permitted in source.
+The type declaration is synthesized by the compiler if not already included in the compilation.
+
+Each type reference in metadata may have an associated `NullableAttribute` with a `byte[]` where each `byte`
+represents nullability: 0 for oblivious, 1 for not annotated, and 2 for annotated.
+
+The `byte[]` is constructed as follows:
+- Reference type: the nullability (0, 1, or 2), followed by the representation of the type arguments in order including containing types
+- Nullable value type: the representation of the type argument only
```

Comment:
If we special case Nullable type, it might make sense to special case its argument. From implementation perspective it should be trivial to skip both of them.